tests/adapter/dbt/tests/adapter/utils/test_timestamps.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+import re
+from dbt.tests.util import check_relation_has_expected_schema, run_dbt
+
+_MODEL_CURRENT_TIMESTAMP = """
+select {{ current_timestamp() }} as current_timestamp,
+       {{ current_timestamp_in_utc_backcompat() }} as current_timestamp_in_utc_backcompat,
+       {{ current_timestamp_backcompat() }} as current_timestamp_backcompat
+"""
+
+_MODEL_EXPECTED_SQL = """
+select now() as current_timestamp,
+       (current_timestamp at time zone 'utc')::TIMESTAMP as current_timestamp_in_utc_backcompat,
+       current_timestamp::TIMESTAMP as current_timestamp_backcompat
+"""
+
+
+class TestCurrentTimestamps:
```

Comment:
@gshank could we get your take on something?

The patten I'm used to seeing is like [this](https://github.com/dbt-labs/dbt-core/blob/c2ab2971b073245366f1ccd8a9e1cbe198f18865/tests/adapter/dbt/tests/adapter/grants/test_invalid_grants.py):

```
import pytest
from dbt.tests.adapter.category.base_thing import BaseThing

class BaseSpecificThing(BaseThing):
    <a bunch of code>

class TestSpecificThing(BaseSpecificThing):
    pass
```

Is there any reason not to skip the last two lines and just do the following instead?
```
import pytest
from dbt.tests.adapter.category.base_thing import BaseThing

class TestSpecificThing(BaseThing):
    <a bunch of code>
```

Reply:
the only difference this makes is that if you import the base it pytest won't run it until the inheriting class is called because it doesn't have "test" in the class name.

Reply:
We can't import test classes with "Test" in their name from the adapter zone, because pytest will run both the imported test and the subclass, which really won't work.  In the specific test files there's no reason to do yet another layer of subclassing. Because you haven't said where the example files are located, I'm a little unclear as to which case you're illustrating. tl;dr:  nothing importable in a different adapter should start with "Test".

Reply:
Took a look at the code, and you'll need to follow the Base<test_name>. Test(Base<test_name): pass pattern.

Reply:
if you inherit as <module name>.<base test class name> it won't run the test twice

Reply:
At this point I think that it's more important to be consistent, so implementers of these tests don't have to do it in multiple ways. If you think we should switch everything over to inheriting from <module name>.<base test class name>, open a ticket for that and it can be discussed.

Reply:
yea that's a fair point, I've changed it to follow the standard pattern